### ISSUE TITLE
How to get the precision as high as you proposed in your benchmark when testing your model?

### ISSUE THREAD

**Issue by cecilia930426** [open-mmlab/mmaction2#265]
I tested the model tsn_r50_320p_1x1x8_50e_activitynet_video_rgb in your benchmark. And the precision is not as good as you proposed. Your proposed top 1 acc is 73.97, however, what I got is only top 1 acc 67.25, top 5 acc 89.93. Could you please help me to solve the problem?
Thanks in advance!

<img width="766" alt="WX20201019-165359@2x" src="https://user-images.githubusercontent.com/16661965/96660077-a4cc6900-1316-11eb-802f-120fea90655c.png">
<img width="762" alt="WX20201019-165329@2x" src="https://user-images.githubusercontent.com/16661965/96660087-a8f88680-1316-11eb-8e10-dfb7494f5721.png">

**Comment by kennymckormick** [comment]
That large difference seems weird, Here are several points that you can check:
1. The reported score is tested on the ActivityNet validation set, our version consists of 4726 raw videos. [However, missing several videos should not make such large difference ....]
2. Did you follow the data preparation protocol list in `tools/data/activitynet`? For example, we process each raw video into 340x256 frames, and generate the file list, each line follows the format: `frame_dir num_frames label`.
3. Did you follow the testing protocol in `tsn_r50_320p_1x1x8_50e_activitynet_video_rgb`. During testing, we uniformly sample 25 frames for each raw video, we use ` ThreeCrop` for augmentation.

Please check if the above settings are aligned with ours and let me know. I can provide the testing log and result file if you need them.

**Comment by cecilia930426** [comment]
> That large difference seems weird, Here are several points that you can check:
> 
> 1. The reported score is tested on the ActivityNet validation set, our version consists of 4726 raw videos. [However, missing several videos should not make such large difference ....]
> 2. Did you follow the data preparation protocol list in `tools/data/activitynet`? For example, we process each raw video into 340x256 frames, and generate the file list, each line follows the format: `frame_dir num_frames label`.
> 3. Did you follow the testing protocol in `tsn_r50_320p_1x1x8_50e_activitynet_video_rgb`. During testing, we uniformly sample 25 frames for each raw video, we use ` ThreeCrop` for augmentation.
> 
> Please check if the above settings are aligned with ours and let me know. I can provide the testing log and result file if you need them.

Could you please tell me your specific sampling strages? Original videos vary from 5fps to 30fps. It is common practice to convert them to some fixed fps. But from your code, I think you did not use a fixed fps. Could you please tell me in detail? 
Thank you.

**Comment by cecilia930426** [comment]
@kennymckormick Another question is that did you change FPS when you convert videos to frames? Could we get the suitable result if we reproducing your implementation by fixed 15FPS?

**Comment by kennymckormick** [comment]
> @kennymckormick Another question is that did you change FPS when you convert videos to frames? Could we get the suitable result if we reproducing your implementation by fixed 15FPS?

Actually, most videos in Kinetics-400 are 30FPS. I think it would be better for you to adjust the FPS of all videos to 30FPS.

**Comment by cecilia930426** [comment]
> > @kennymckormick Another question is that did you change FPS when you convert videos to frames? Could we get the suitable result if we reproducing your implementation by fixed 15FPS?
> 
> Actually, most videos in Kinetics-400 are 30FPS. I think it would be better for you to adjust the FPS of all videos to 30FPS.

I mean ActivityNet 1.3

**Comment by kennymckormick** [comment]
> > > @kennymckormick Another question is that did you change FPS when you convert videos to frames? Could we get the suitable result if we reproducing your implementation by fixed 15FPS?
> > 
> > 
> > Actually, most videos in Kinetics-400 are 30FPS. I think it would be better for you to adjust the FPS of all videos to 30FPS.
> 
> I mean ActivityNet 1.3

We didn't change the FPS when preparing frames. If you change it to 15fps, I think it's no big deal, since we extract features from the raw video sparsely. 
There's another thing that you need to check: we use models trained on Kinetics-400 for initialization (also written in the config file). Did you initialize the model with Kinetics pre-trained weights before training starts?

**Comment by cecilia930426** [comment]
> > > > @kennymckormick Another question is that did you change FPS when you convert videos to frames? Could we get the suitable result if we reproducing your implementation by fixed 15FPS?
> > > 
> > > 
> > > Actually, most videos in Kinetics-400 are 30FPS. I think it would be better for you to adjust the FPS of all videos to 30FPS.
> > 
> > 
> > I mean ActivityNet 1.3
> 
> We didn't change the FPS when preparing frames. If you change it to 15fps, I think it's no big deal, since we extract features from the raw video sparsely.
> There's another thing that you need to check: we use models trained on Kinetics-400 for initialization (also written in the config file). Did you initialize the model with Kinetics pre-trained weights before training starts?

You mean I need to do it when testing your benchmark model? Or I should use your code to train my own model if I change FPS to a fixed15fps? And I could not get the same results as yours when I testing your benchmark model by changing it to 15 fps? Thanks!

**Comment by kennymckormick** [comment]
I think it's OK to test my model on 15fps video. As mentioned in the config file, to test ` tsn_r50_320p_1x1x8_50e_activitynet_video_rgb` you should sample 25 frames uniformly from the untrimmed video with `ThreeCrop`. I think you can attach your: 1. testing config, 2. testing log, 3. testing file list 4. testing results (maybe around several megabytes in total), for benchmark model testing to help me better locate the problem.

**Comment by kennymckormick** [comment]
> > > > > @kennymckormick Another question is that did you change FPS when you convert videos to frames? Could we get the suitable result if we reproducing your implementation by fixed 15FPS?
> > > > 
> > > > 
> > > > Actually, most videos in Kinetics-400 are 30FPS. I think it would be better for you to adjust the FPS of all videos to 30FPS.
> > > 
> > > 
> > > I mean ActivityNet 1.3
> > 
> > 
> > We didn't change the FPS when preparing frames. If you change it to 15fps, I think it's no big deal, since we extract features from the raw video sparsely.
> > There's another thing that you need to check: we use models trained on Kinetics-400 for initialization (also written in the config file). Did you initialize the model with Kinetics pre-trained weights before training starts?
> 
> You mean I need to do it when testing your benchmark model? Or I should use your code to train my own model if I change FPS to a fixed15fps? And I could not get the same results as yours when I testing your benchmark model by changing it to 15 fps? Thanks!

I think it's OK to test my model on 15fps video. As mentioned in the config file, to test ` tsn_r50_320p_1x1x8_50e_activitynet_video_rgb` you should sample 25 frames uniformly from the untrimmed video with `ThreeCrop`. I think you can attach your: 1. testing config, 2. testing log, 3. testing file list 4. testing results (maybe around several megabytes in total), for benchmark model testing to help me better locate the problem.
